versions.tf

```diff
@@ -7,35 +7,35 @@ terraform {
   required_providers {
     google = {
       source  = "hashicorp/google"
-      version = "4.63.1"
+      version = "5.12.0"
     }
     google-beta = {
       source  = "hashicorp/google-beta"
-      version = "4.63.1"
+      version = "5.12.0"
     }
     kubernetes = {
       source  = "hashicorp/kubernetes"
-      version = "2.20.0" # Constrained by Google
+      version = "~> 2.25.2" # Constrained by Google
```

Comment:
These are all patch level versions. Just checking to be sure that's what's required here. Seems like the team has been moving toward minor updates like "~> 2.25" unless these values are still truly constrained by google. This comment would apply to **ALL** versions being adjusted here.

Reply:
I purposely made it so that the patch version could be >= to the last number. I havn't seen it in the kubernetes provider in particular, but in Google case some providers/modules introduce breaking changes even during minor version bumps which can be problematic.

See terraform-provider-google:5.4.0 changelog as a quick example https://github.com/hashicorp//blob/main/CHANGELOG.md#540-oct-30-2023

I just found it to be safest in this project's case to only allow variations in the patch.

Reply:
Ok, I'll approve but believe that David has some other thoughts on this one. Might be a good time to have a team meeting about versioning ;)